actions/databases.go

```diff
@@ -0,0 +1,290 @@
+//   If the `all=true` parameter is passed it will return a list of clusters in addition to instances.
+func DatabasesList(c buffalo.Context) error {
+	// if all param is given, we'll return information about both instances and clusters
+	// otherwise, only database instances will be returned
+	all := false
+	if b, err := strconv.ParseBool(c.Param("all")); err == nil {
+		all = b
+	}
+
+	rdsClient := RDS[c.Param("account")]
+	var clustersOutput *rds.DescribeDBClustersOutput
+	var instancesOutput *rds.DescribeDBInstancesOutput
+	var err error
+
+	if all {
+		clustersOutput, err = rdsClient.Service.DescribeDBClusters(&rds.DescribeDBClustersInput{})
+		if err != nil {
+			log.Println(err.Error())
+		}
+	}
+
+	instancesOutput, err = rdsClient.Service.DescribeDBInstances(&rds.DescribeDBInstancesInput{})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			log.Println(aerr.Error())
+			return c.Error(400, aerr)
+		} else {
+			log.Println(err.Error())
+			return err
+		}
+	}
+
+	output := struct {
+		*rds.DescribeDBClustersOutput
+		*rds.DescribeDBInstancesOutput
+	}{
+		clustersOutput,
+		instancesOutput,
+	}
+	return c.Render(200, r.JSON(output))
+}
+
+// DatabasesGet gets details about a specific database
+//   If the `all=true` parameter is passed it will return a list of clusters in addition to instances.
+func DatabasesGet(c buffalo.Context) error {
+	// if all param is given, we'll return information about both instances and clusters
+	// otherwise, only database instances will be searched
+	all := false
+	if b, err := strconv.ParseBool(c.Param("all")); err == nil {
+		all = b
+	}
+
+	rdsClient := RDS[c.Param("account")]
+	var clustersOutput *rds.DescribeDBClustersOutput
+	var instancesOutput *rds.DescribeDBInstancesOutput
+	var err error
+	clusterNotFound := true
+
+	if all {
+		// search clusters for the given db name
+		clusterNotFound = false
+		clustersInput := &rds.DescribeDBClustersInput{
+			DBClusterIdentifier: aws.String(c.Param("db")),
+		}
+		clustersOutput, err = rdsClient.Service.DescribeDBClusters(clustersInput)
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				if aerr.Code() == rds.ErrCodeDBClusterNotFoundFault {
+					log.Println(rds.ErrCodeDBClusterNotFoundFault, aerr.Error())
+					clusterNotFound = true
+				} else {
+					log.Println(aerr.Error())
+				}
+			} else {
+				log.Println(err.Error())
+			}
+		}
+	}
+
+	// search instances for the given db name
+	instancesInput := &rds.DescribeDBInstancesInput{
+		DBInstanceIdentifier: aws.String(c.Param("db")),
+	}
+	instancesOutput, err = rdsClient.Service.DescribeDBInstances(instancesInput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == rds.ErrCodeDBInstanceNotFoundFault {
+				log.Println(rds.ErrCodeDBInstanceNotFoundFault, aerr.Error())
+				if clusterNotFound {
+					return c.Error(404, aerr)
+				}
+			} else {
+				log.Println(aerr.Error())
+				return c.Error(400, aerr)
```

Comment:
why is this a 400?  i think this else can go away

Reply:
without this our API will return a 500 when the RDS API returns 400

Reply:
ok - the aws error parsing stuff seems so verbose but I guess it's okay